filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest_test.go

```diff
@@ -750,98 +802,126 @@ func BenchmarkAuthorizeRequest(b *testing.B) {
 						payload.sub == "5974934733"
 					}				
 				`, publicKey),
-			}),
-		)
-
-		f, err := createOpaFilter(opaControlPlane)
-		assert.NoError(b, err)
-
-		url, err := url.Parse("http://opa-authorized.test/somepath")
-		assert.NoError(b, err)
-
-		claims := jwt.MapClaims{
-			"iss":   "https://some.identity.acme.com",
-			"sub":   "5974934733",
-			"aud":   "nqz3xhorr5",
-			"iat":   time.Now().Add(-time.Minute).UTC().Unix(),
-			"exp":   time.Now().Add(tokenExp).UTC().Unix(),
-			"email": "someone@example.org",
-		}
+		}),
+	)
 
-		token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	decisionLogConsumer := newDecisionConsumer()
+	defer decisionLogConsumer.Close()
 
-		privKey, err := os.ReadFile(keyPath)
-		if err != nil {
-			log.Fatalf("Failed to read priv key: %v", err)
-		}
+	f, err := createOpaFilter(opaControlPlane, decisionLogConsumer, decisionLogging)
+	assert.NoError(b, err)
 
-		key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privKey))
-		if err != nil {
-			log.Fatalf("Failed to parse RSA PEM: %v", err)
-		}
+	url, err := url.Parse("http://opa-authorized.test/somepath")
+	assert.NoError(b, err)
 
-		// Sign and get the complete encoded token as a string using the secret
-		signedToken, err := token.SignedString(key)
-		if err != nil {
-			log.Fatalf("Failed to sign token: %v", err)
-		}
+	claims := jwt.MapClaims{
+		"iss":   "https://some.identity.acme.com",
+		"sub":   "5974934733",
+		"aud":   "nqz3xhorr5",
+		"iat":   time.Now().Add(-time.Minute).UTC().Unix(),
+		"exp":   time.Now().Add(tokenExp).UTC().Unix(),
+		"email": "someone@example.org",
+	}
 
-		ctx := &filtertest.Context{
-			FStateBag: map[string]interface{}{},
-			FResponse: &http.Response{},
-			FRequest: &http.Request{
-				Header: map[string][]string{
-					"Authorization": {fmt.Sprintf("Bearer %s", signedToken)},
-				},
-				URL: url,
-			},
-			FMetrics: &metricstest.MockMetrics{},
-		}
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+
+	privKey, err := os.ReadFile(keyPath)
+	if err != nil {
+		log.Fatalf("Failed to read priv key: %v", err)
```

Comment:
Why not require.NoError(t, err) here and below?

Reply:
would be good to also address this @mefarazath